PayloadPkg/OsLoader/LoadImage.c

```diff
@@ -1,571 +1,579 @@
-  FileSize   = 0;
-  FileBuffer = NULL;
-  FileHandle = NULL;
-  Status = OpenFile (FsHandle, FileName, &FileHandle);
-  if (EFI_ERROR (Status)) {
-    DEBUG ((DEBUG_INFO, "Open file '%s' failed, Status = %r\n", FileName, Status));
-    goto Done;
-  }
-
-  Status = GetFileSize (FileHandle, &FileSize);
-  if (EFI_ERROR (Status)) {
-    DEBUG ((DEBUG_INFO, "Get file '%s' size failed, Status = %r\n", FileName, Status));
-    goto Done;
-  }
-
-  FileBuffer = AllocatePool (FileSize);
-  if (FileBuffer == NULL) {
-    Status = EFI_OUT_OF_RESOURCES;
-    goto Done;
-  }
-
-  Status = ReadFile (FileHandle, &FileBuffer, &FileSize);
-  DEBUG ((DEBUG_INFO, "Load file %a [size %d bytes]: %r\n", Ptr, FileSize, Status));
-  if (!EFI_ERROR (Status)) {
-    ImageData->Addr = FileBuffer;
-    ImageData->Size = FileSize;
-  } else {
-    if (FileBuffer != NULL) {
-      FreePool (FileBuffer);
-    }
-  }
-
-Done:
-  if (FileHandle != NULL) {
-    CloseFile (FileHandle);
-  }
-  return Status;
-}
-
-/**
-  Get traditional linux image from file
-
-  This function will read traditional linux files (vmlinuz, config.cfg
-  and initrd) from file system into memory. The loaded file info will
-  be saved to LinuxImage.
-
-  @param[in]  FsHandle        File system handle used to read file
-  @param[out] LinuxImage      Used to save loaded Image information.
-
-  @retval  RETURN_SUCCESS     If IAS image was loaded successfully
-  @retval  Others             If IAS image was not loaded.
-**/
-EFI_STATUS
-GetTraditionalLinux (
-  IN  EFI_HANDLE             FsHandle,
-  OUT LINUX_IMAGE            *LinuxImage
-  )
-{
-  RETURN_STATUS              Status;
-  UINTN                      ConfigFileSize;
-  VOID                       *ConfigFile;
-  LINUX_BOOT_CFG             LinuxBootCfg;
-  UINT32                     Index;
-  UINT32                     EntryIdx;
-  CHAR8                      *Ptr;
-  MENU_ENTRY                 *MenuEntry;
-  EFI_HANDLE                 FileHandle;
-  BOOLEAN                    DefBootOption;
-
-  ConfigFile     = NULL;
-  ConfigFileSize = 0;
-  DefBootOption  = FALSE;
-  Status = RETURN_NOT_FOUND;
-
-  DEBUG ((DEBUG_INFO, "Try booting Linux from config file ...\n"));
-
-  for (Index = 0; Index < (UINTN)(FeaturePcdGet (PcdGrubBootCfgEnabled) ? 2 : 1); Index++) {
-    DEBUG ((DEBUG_INFO, "Checking %s\n",mConfigFileName[Index]));
-    ConfigFile     = NULL;
-    ConfigFileSize = 0;
-
-    Status = OpenFile (FsHandle, (CHAR16 *)mConfigFileName[Index], &FileHandle);
-    if (EFI_ERROR (Status)) {
-      DEBUG ((DEBUG_INFO, "Open file '%s' failed, Status = %r\n", (CHAR16 *)mConfigFileName[Index], Status));
-      continue;
-    }
-
-    Status = GetFileSize (FileHandle, &ConfigFileSize);
-    if (EFI_ERROR (Status)) {
-      DEBUG ((DEBUG_INFO, "Get file '%s' size failed, Status = %r\n", (CHAR16 *)mConfigFileName[Index], Status));
-      CloseFile (FileHandle);
-      continue;
-    }
-
+/** @file
```

Comment:
It looks line ending changed. Can you keep previous line ending feed for better code review? Let's clean up line ending issue later in entire files.